buildSrc/src/main/groovy/org/mockito/release/notes/vcs/DefaultContributionSet.java

```diff
@@ -20,12 +22,29 @@ void add(Commit commit) {
         }
         commits.add(commit);
         tickets.addAll(commit.getTickets());
-        Contribution c = contributions.get(commit.getAuthorId());
-        if (c == null) {
-            contributions.put(commit.getAuthorId(), new Contribution(commit));
+
+        Contribution existing = findContribution(commit, contributions);
+        if (existing != null) {
+            existing.add(commit);
         } else {
-            c.add(commit);
+            contributions.add(new Contribution(commit));
+        }
+    }
+
+    private static Contribution findContribution(Commit commit, Iterable<Contribution> contributions) {
+        for (Contribution c : contributions) {
+            //From Git Log we don't know the GitHub user ID, only the email and name.
+            //Sometimes contributors have different email addresses while the same name
+            //This leads to awkward looking release notes, where same author is shown multiple times
+            //We consider the contribution to be the same if any of: email or name is the same
+            //
+            //This approach comes with a caveat. What if the user have same author name, different email and indeed it is a different user?
+            // This scenario is not handled well but it is unlikely and we consider it a trade-off
```

Comment:
Thanks for hint and review!!!

I didn't know about this command. I've tried it but it does not give me github user id. However, it does seem interesting and it solves the problem of misspelled names / emails. Wanna help, PR! :)